src/test/java/edu/unc/lib/boxc/migration/cdm/services/FieldUrlAssessmentServiceTest.java

```diff
@@ -0,0 +1,304 @@
+        stubFor(get(urlEqualTo( "/new_url_caption_again"))
+                .willReturn(aResponse()
+                        .withStatus(statusCode)));
+    }
+
+    private void stubRedirectUrls() {
+        stubFor(get(urlEqualTo("/00276/"))
+                .willReturn(permanentRedirect(cdmBaseUrl + "/new_redirect_url")));
+        stubFor(get(urlEqualTo("/new_url_notes"))
+                .willReturn(permanentRedirect(cdmBaseUrl + "/new_redirect_url")));
+        stubFor(get(urlEqualTo("/new_url_description"))
+                .willReturn(permanentRedirect(cdmBaseUrl + "/new_redirect_url")));
+        stubFor(get(urlEqualTo("/new_url_caption"))
+                .willReturn(permanentRedirect(cdmBaseUrl + "/new_redirect_url")));
+        stubFor(get(urlEqualTo("/new_url_caption_again"))
+                .willReturn(permanentRedirect(cdmBaseUrl + "/new_redirect_url")));
+        stubFor(get(urlEqualTo("/new_redirect_url"))
+                .willReturn(permanentRedirect(cdmBaseUrl + "/final_redirect_url")));
+        stubFor(get(urlEqualTo( "/final_redirect_url"))
+                .willReturn(aResponse()
+                        .withStatus(HttpStatus.SC_OK)));
+    }
+
+    private void addUrlsToDb() throws SQLException {
+        Connection conn = indexService.openDbConnection();
+        Statement stmt = conn.createStatement();
+        stmt.executeUpdate("UPDATE " + CdmIndexService.TB_NAME + " SET descri = descri || '" +
+                cdmBaseUrl + "/new_url_description' WHERE " + CdmFieldInfo.CDM_ID + " = 25");
+        stmt.executeUpdate("UPDATE " + CdmIndexService.TB_NAME + " SET descri = descri || '" +
+                cdmBaseUrl + "/new_url_description' WHERE " + CdmFieldInfo.CDM_ID + " = 26");
+        stmt.executeUpdate("UPDATE " + CdmIndexService.TB_NAME + " SET notes = notes || '" +
+                cdmBaseUrl + "/new_url_notes' WHERE " + CdmFieldInfo.CDM_ID + "= 28");
+        stmt.executeUpdate("UPDATE " + CdmIndexService.TB_NAME + " SET captio = captio || '" +
+                cdmBaseUrl + "/new_url_caption' WHERE " + CdmFieldInfo.CDM_ID + " = 25");
+        stmt.executeUpdate("UPDATE " + CdmIndexService.TB_NAME + " SET captio = captio || '" +
+                cdmBaseUrl + "/new_url_caption_again' WHERE " + CdmFieldInfo.CDM_ID + " = 26");
+        stmt.executeUpdate("UPDATE " + CdmIndexService.TB_NAME + " SET source = 'Jeremy Francis Gilmer Papers; " +
+                cdmBaseUrl + "/00276/'");
+        CdmIndexService.closeDbConnection(conn);
+    }
+
+    private void addProblematicUrlToDb(String url) throws SQLException {
+        Connection conn = indexService.openDbConnection();
+        Statement stmt = conn.createStatement();
+        stmt.executeUpdate("UPDATE " + CdmIndexService.TB_NAME + " SET descri = descri || '" +
+                url + "' WHERE " + CdmFieldInfo.CDM_ID + " = 100");
+        CdmIndexService.closeDbConnection(conn);
+    }
+
+    private void assertUrlsInCsvAreCorrect(List<CSVRecord> rows) {
```

Comment:
I might have missed it, but I don't think we are verify that the NICK_FIELD column is being populated anywhere in the test. Could probably incorporate it here.